password-store-1.7.3 fails to build on darwin

## Issue description

The installation of `password-store-1.7.3`

### Steps to reproduce

```sh
$ nix-env -i password-store-1.7.3
```

## Technical details

Output of the command:

```
warning: there are multiple derivations named 'password-store-1.7.3'; using the first one
installing 'password-store-1.7.3'
these derivations will be built:
  /nix/store/dzfphsgmc4c65dgkz4pj4z0w1k85cqdk-password-store-1.7.3.drv
building '/nix/store/dzfphsgmc4c65dgkz4pj4z0w1k85cqdk-password-store-1.7.3.drv'...
unpacking sources
unpacking source archive /nix/store/36dqisv8vdb1dryplh4zxsqif9bsqhl2-password-store-1.7.3.tar.xz
source root is password-store-1.7.3
setting SOURCE_DATE_EPOCH to timestamp 1533267168 of file password-store-1.7.3/tests/t0500-find.sh
patching sources
applying patch /nix/store/k4mklq09ga6jqqjh8ivzcbna42yjwq7k-set-correct-program-name-for-sleep.patch
patching file src/password-store.sh
patching file src/platform/cygwin.sh
patching file src/platform/darwin.sh
applying patch /nix/store/0rviz57x7x65g1ij97ibi5fjw7pf37hj-no-darwin-getopt.patch
patching file src/platform/darwin.sh
Hunk #1 succeeded at 43 with fuzz 2 (offset 12 lines).
patching script interpreter paths in tests
tests/fake-editor-change-password.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0001-sanity-checks.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0010-generate-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0020-show-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0050-mv-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0060-rm-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0100-insert-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0200-edit-tests.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0300-reencryption.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0400-grep.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
tests/t0500-find.sh: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash
Password store is a shell script, so there is nothing to do. Try "make install" instead.
installing
install flags: SHELL=/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash PREFIX=\$\(out\) WITH_ALLCOMP=yes install
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/man'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/man/man1'
'man/pass.1' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/man/man1/pass.1'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/bash-completion'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/bash-completion/completions'
'src/completion/pass.bash-completion' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/bash-completion/completions/pass'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/zsh'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/zsh/site-functions'
'src/completion/pass.zsh-completion' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/zsh/site-functions/_pass'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/fish'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/fish/vendor_completions.d'
'src/completion/pass.fish-completion' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/fish/vendor_completions.d/pass.fish'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/lib'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/lib/password-store'
'src/platform/darwin.sh' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/lib/password-store/platform.sh'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/lib/password-store/extensions'
install: creating directory '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/bin/'
'src/.pass' -> '/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/bin/pass'
post-installation fixup
gzipping man pages under /nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/share/man/
strip is /nix/store/3ibh116gnggzgsvrq8vvalmqyipk8cy9-cctools-binutils-darwin/bin/strip
stripping (with command strip and flags -S) in /nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/lib  /nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/bin 
patching script interpreter paths in /nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3
/nix/store/xs9vcl7c8m4as2gsddfknyslby5fq32c-password-store-1.7.3/bin/pass: interpreter directive changed from "/usr/bin/env bash" to "/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash"
running install tests
installcheck flags: SHELL=/nix/store/06z61jbgs0vkw4i9cqqf9yl7zsfkkhw2-bash-4.4-p23/bin/bash test
ok 1 - Make sure we can run pass
ok 2 - Make sure we can initialize our test store
# passed all 2 test(s)
1..2
not ok 1 - Test "generate" command
#       
#               "$PASS" init $KEY1 &&
#               "$PASS" generate cred 19 &&
#               [[ $("$PASS" show cred | wc -m) -eq 20 ]]
#       
not ok 2 - Test replacement of first line
#       
#               "$PASS" insert -m cred2 <<<"$(printf "this is a big\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" &&
#               "$PASS" generate -i cred2 23 &&
#               [[ $("$PASS" show cred2) == "$(printf "%s\\npassword\\nwith\\nmany\\nlines\\nin it bla bla" "$("$PASS" show cred2 | head -n 1)")" ]]
#       
# failed 2 among 2 test(s)
1..2
make: *** [Makefile:72: tests/t0010-generate-tests.sh] Error 1
builder for '/nix/store/dzfphsgmc4c65dgkz4pj4z0w1k85cqdk-password-store-1.7.3.drv' failed with exit code 2
error: build of '/nix/store/dzfphsgmc4c65dgkz4pj4z0w1k85cqdk-password-store-1.7.3.drv' failed
```

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-darwin"`
 - host os: `Darwin 16.7.0, macOS 10.12.6`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.2.1`
 - channels(splinter): `"darwin, nixpkgs-19.09pre173445.796a8764ab8"`
 - nixpkgs: `/Users/splinter/.nix-defexpr/channels/nixpkgs`
```

I'm running into a suspiciously similar issue:

I'm trying to install pass-otp using nix `pass.withExtensions (es: [ es.pass-otp ])`, this works on two linux boxes and one
mac but not the other. Seemingly because of test failures but I'm not sure why
the tests are failing.

Further digging has revealed that the failing test is one of the tests for
regular pass and that pass installs fine if I drop the pass-otp extension. I
still don't have any idea why it works fine with the extension on another mac.

```
ok 1 - Make sure we can run pass
ok 2 - Make sure we can initialize our test store
# passed all 2 test(s)
1..2
not ok 1 - Test "generate" command
#       
#               "$PASS" init $KEY1 &&
#               "$PASS" generate cred 19 &&
#               [[ $("$PASS" show cred | wc -m) -eq 20 ]]
#       
not ok 2 - Test replacement of first line
#       
#               "$PASS" insert -m cred2 <<<"$(printf "this is a big\\npassword\\nwith\\nmany\\nlines\\nin it bla bla")" &&
#               "$PASS" generate -i cred2 23 &&
#               [[ $("$PASS" show cred2) == "$(printf "%s\\npassword\\nwith\\nmany\\nlines\\nin it bla bla" "$("$PASS" show cred2 | head -n 1)")" ]]
#       
# failed 2 among 2 test(s)
1..2
make: *** [Makefile:72: tests/t0010-generate-tests.sh] Error 1
builder for '/nix/store/3h3w38pganm4jzws1wv0lp91wn01q7r1-password-store-1.7.3.drv' failed with exit code 2
```

--- a/pkgs/tools/security/pass/default.nix
+++ b/pkgs/tools/security/pass/default.nix
@@ -111,6 +111,12 @@ let
     '' + stdenv.lib.optionalString stdenv.isDarwin ''
       # 'pass edit' uses hdid, which is not available from the sandbox.
       rm -f tests/t0200-edit-tests.sh
+      rm -f tests/t0010-generate-tests.sh
+      rm -f tests/t0020-show-tests.sh
+      rm -f tests/t0050-mv-tests.sh
+      rm -f tests/t0100-insert-tests.sh
+      rm -f tests/t0300-reencryption.sh
+      rm -f tests/t0400-grep.sh
     '';
 
     doCheck = false;

